docs/best-practices.md

```diff
@@ -0,0 +1,28 @@
+Best Practices
+---------
+
+This page is a placeholder for best practices we recommend you to review while implementing Vue InstantSearch.
```

Comment:
The updates to this content / new page is great! Do we have any app-level best practices we'd like to offer as well?